src/machine/predictor.h

```diff
@@ -99,162 +103,144 @@ struct PredictionFeedback {
-        uint8_t number_of_bht_bits,
-        PredictorState initial_state);
-
-protected: // Internal functions
-    uint8_t init_number_of_bht_addr_bits(uint8_t b) const;
-    uint8_t init_number_of_bht_bits(uint8_t b) const;
-    uint16_t calculate_bht_index(const uint16_t bhr_value, const Address instruction_address) const;
-    void update_stats(PredictionFeedback feedback) override;
-    BranchResult convert_state_to_prediction(PredictorState state) const;
-    virtual void update_bht(PredictionFeedback feedback) = 0;
-
-public:                                                   // General functions
-    BranchResult predict(PredictionInput input) override; // Function which handles all actions ties
-                                                          // to making a branch prediction
-    void update(PredictionFeedback feedback) override;    // Update predictor based on jump / branch
-                                                          // result
-
-protected:                                 // Internal variables
-    const uint8_t number_of_bht_addr_bits; // Number of Branch History Table (BHT) bits taken from
-                                           // instruction address
-    const uint8_t number_of_bht_bits;      // Number of Branch History Table (BHT) bits
-    std::vector<BranchHistoryTableEntry> bht; // Branch History Table (BHT)
+    bool is_static() const override { return true; };
+    BranchResult predict(PredictionInput input) override;
+    void update(PredictionFeedback feedback) override;
 };
 
 // Dynamic Predictor - Smith 1 Bit
-class PredictorSmith1Bit final : public PredictorSmith {
+class PredictorSmith1Bit final : public Predictor {
 public: // Constructors & Destructor
     PredictorSmith1Bit(
         uint8_t number_of_bht_addr_bits,
         uint8_t number_of_bht_bits,
         PredictorState initial_state);
 
-private: // Internal functions
-    BranchResult make_prediction(PredictionInput input) const override;
-    void update_bht(PredictionFeedback feedback);
-
 public: // General functions
     PredictorType get_type() const override { return PredictorType::SMITH_1_BIT; };
+    bool is_static() const override { return false; };
+    BranchResult predict(PredictionInput input) override;
+    void update(PredictionFeedback feedback) override;
 };
 
 // Dynamic Predictor - Smith 2 Bit
-class PredictorSmith2Bit final : public PredictorSmith {
+class PredictorSmith2Bit final : public Predictor {
 public: // Constructors & Destructor
     PredictorSmith2Bit(
         uint8_t number_of_bht_addr_bits,
         uint8_t number_of_bht_bits,
         PredictorState initial_state);
 
-private: // Internal functions
-    BranchResult make_prediction(PredictionInput input) const override;
-    void update_bht(PredictionFeedback feedback);
-
 public: // General functions
     PredictorType get_type() const override { return PredictorType::SMITH_2_BIT; };
+    bool is_static() const override { return false; };
```

Comment:
If we have this method, why is GUI using manual check for dynamic using enum?

Reply:
Moved this to separate function